platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConfigDataFactory.java

```diff
@@ -89,34 +95,66 @@ private Object getValueForRecordComponent(final String namePrefix, final RecordC
                 final Class<?> genericType = getGenericListType(component);
                 return configuration.getValues(name, genericType, getDefaultValues(component));
             }
+            if (Objects.equals(Set.class, component.getType())) {
+                final Class<?> genericType = getGenericSetType(component);
+                return configuration.getValueSet(name, genericType, getDefaultValueSet(component));
+            }
             return configuration.getValue(name, valueType, getDefaultValue(component));
         } else {
             if (Objects.equals(List.class, component.getType())) {
                 final Class<?> genericType = getGenericListType(component);
                 return configuration.getValues(name, genericType);
             }
+            if (Objects.equals(Set.class, component.getType())) {
+                final Class<?> genericType = getGenericSetType(component);
+                return configuration.getValueSet(name, genericType);
+            }
             return configuration.getValue(name, valueType);
         }
     }
 
+    private static boolean isGenericType(@NonNull final RecordComponent component, @NonNull final Class<?> type) {
```

Comment:
Is `isGenericType` the right name for this predicate?  Seem more like `IsGenericTypeOfThisBaseType` or something ...

Whether it is or isn't, perhaps some javadoc to describe what the parameters are?

Reply:
good point. Will put it on my list (do not want to change this PR anymore since it already has several approvals)